code/modules/antagonists/eldritch_cult/eldritch_monster_antag.dm

```diff
@@ -0,0 +1,43 @@
+///Tracking reasons
+/datum/antagonist/heretic_monster
+	name = "Eldritch Horror"
+	roundend_category = "Heretics"
+	antagpanel_category = "Heretic Beast"
+	antag_moodlet = /datum/mood_event/heretics
+	job_rank = ROLE_HERETIC
+	antag_hud_type = ANTAG_HUD_HERETIC
+	antag_hud_name = "heretic_beast"
+	var/mob/living/master
```

Comment:
The master should be the heretic antag datum not the mob.

Reply:
making it a mob is more modular. What if some admin wants to play a worm summoner and spawns in a few armsies and binds them to himself? The mobs should still be allied to you even if you somehow stop being a heretic

Reply:
eh whatever i will replace it, admins can just make themselves into heretics not a big deal

Reply:
Mob is the opposite of modular, considering the mob changes if you turn yourself into armsy.
You could also update the master mob whenever the minds mob changes but that sounds like a lot more work.

If an admin wants to do this, they can do it properly by actually giving themselves the antag datum.

The main problem with mob is that it can be deleted, so this value can become null and would force a hard delete (if mobs weren't doing that anyways). At least bind it to mind.

Reply:
i did what you suggested and just bound it to antag datum, after thinking i came to the same conclusion as you did

Reply:
anyways this got fixed i will be resolving this